Mathlib/Tactic/ShowTerm.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Scott Morrison
+-/
+import Lean.Elab.Tactic.Basic
+
+open Lean
+open Lean.Meta
+open Lean.Elab.Tactic
+
+/-
+TODO: Support for metavariables
+
+The `show_term` tactic is meant to cope with metavariables,
+and be able to print results like `refine X _ Y`.
+
+In mathlib we achieve this with the following hack:
+```
+meta def replace_mvars (e : expr) : expr :=
+e.replace (λ e' _, if e'.is_mvar then some (unchecked_cast pexpr.mk_placeholder) else none)
+```
+which replaces all metavariables with underscores.
+
+How would one achieve this in Lean4? Perhaps we're meant to
+* delaborate to a `Syntax`,
+* replace the metavariables with `(_ : $stx)` (perhaps using `Syntax.replaceM`?),
+* and then feed the result into the remainder of the pretty printing pipeline.
+
+The alternatives I can think of are:
+* wish for a `pp.mvarsAsBlanks` option for the pretty printer
+* string munging after pretty printing
+-/
+
+/--
+Construct a `Try this: refine ...` or `Try this: exact ...` string which would construct `g`.
+-/
+def refineString (g : Expr) : TacticM String := do
+  let g₁ ← Expr.headBeta <$> (instantiateMVars g)
+  -- FIXME we need to replace metavariables with underscores here (see note above).
+  let r ← ppExpr g₁
+  if g₁.hasExprMVar then s!"Try this: refine {r}" else s!"Try this: exact {r}"
```

Comment:
I feel like we should be doing the "Try this" properly in Lean 4, instead of using a fragile pile of regexps.  @Kha @Vtec234 Do you have any concrete plans for a "try this" replacement?  Is it already possible to register code actions?